src/main/java/graphql/execution/ExecutionContext.java

```diff
@@ -349,4 +357,33 @@ public ExecutionContext transform(Consumer<ExecutionContextBuilder> builderConsu
     public ResultNodesInfo getResultNodesInfo() {
         return resultNodesInfo;
     }
+
+    @Internal
+    public boolean isRunning() {
+        return isRunning.get() > 0;
+    }
+
+    public <T> T run(Callable<T> callable) {
+        isRunning.incrementAndGet();
+        try {
+            return callable.call();
+        } catch (Exception e) {
```

Comment:
why catch Exception - this seems wrong.

A GraphqlAbort exception for example will be wrapped in RuntimeException and its meaning lost

Ahh Callable throws exception - perhaps make a RuntimeCallable class instead